charts/kong/dispatch-transformer/handler.lua

```diff
@@ -84,15 +89,34 @@ local function substitute_payload(conf, result)
   if data then
     ngx.log(ngx.DEBUG, "request body: " .. data)
   else
-    data = "{}"
+    data = {}
   end
 
-  local ok, json_data = parse_json(data)
-  if ok then
-    ngx.log(ngx.DEBUG, "request body is json")
-    result[conf.substitute.input] = json_data
+  local header = ngx.req.get_headers()
+  if is_json_body(header) then
+    local ok, json_data = parse_json(data)
+    if ok then
+      ngx.log(ngx.DEBUG, "request body is json")
+      result[conf.substitute.input] = json_data
+    else
+      ngx.log(ngx.DEBUG, "request body is not json")
+      result[conf.substitute.input] = data
+    end
+  elseif is_form_urlencoded_body(header) then
+    local form_urlencoded_data, err = ngx.req.get_post_args()
+    if not err then
+      ngx.log(ngx.DEBUG, "request body is x-www-form-urlencoded")
+      result[conf.substitute.input] = form_urlencoded_data
+    else
+      ngx.log(ngx.DEBUG, "request body is not x-www-form-urlencoded: " .. err)
```

Comment:
Again, this seems wrong.

Reply:
looks like error is returned below that? Or is this comment outdated?

Reply:
Yeah that comment is outdated. I returned the error right below that line.